domain-ee/ee-ep-merge-app/src/python_src/service/job_store.py

```diff
@@ -1,17 +1,50 @@
-from pydantic_models import MergeJob
+from db.database import Database, database
+from fastapi.encoders import jsonable_encoder
+from model.merge_job import MergeJob
+from schema import merge_job as schema
 
 
 class JobStore:
-    _JOB_MAP = {}
+
+    def __init__(self, db: Database):
+        self.db = db
+
+    def init(self) -> list[MergeJob]:
+        jobs_in_progress = [schema.MergeJob.model_validate(job) for job in self.get_merge_jobs_in_progress()]
+        jobs_to_restart = []
+        for job in jobs_in_progress:
+            if job.state == schema.JobState.RUNNING_MOVE_CONTENTIONS_TO_PENDING_CLAIM:
+                job.error()
+                self.update_merge_job(job)
+            elif job.state == schema.JobState.RUNNING_CANCEL_EP400_CLAIM or job.state == schema.JobState.RUNNING_ADD_CLAIM_NOTE_TO_EP400:
+                jobs_to_restart.append(job)
+            elif job.state != schema.JobState.COMPLETED_ERROR:
+                job.state = schema.JobState.PENDING
+                self.update_merge_job(job)
+                jobs_to_restart.append(job)
+        return jobs_to_restart
 
     def clear(self):
-        self._JOB_MAP.clear()
+        self.db.clear(MergeJob)
+
+    def get_merge_jobs(self, filter) -> list[MergeJob]:
+        return self.db.query_all(MergeJob, filter)
+
+    def get_all_merge_jobs(self) -> list[MergeJob]:
+        return self.get_merge_jobs(True)
```

Comment:
Would this be more readable if `get_merge_jobs` defaulted to `filter = True` (i.e. no filter)? Because from this line it's not clear what the True being passed in means.

Reply:
It would! Since the stacked [PR](https://github.com/department-of-veterans-affairs/abd-vro/pull/2415) builds on the jobstore code, I'll make the update there!

Reply:
This is updated in the stacked [PR](https://github.com/department-of-veterans-affairs/abd-vro/pull/2415).